homeassistant/components/sensor/buienradar.py

```diff
@@ -0,0 +1,471 @@
+                self._state = new_state
+                self._entity_picture = img
+                return True
+        else:
+            # update all other sensors
+            new_state = data[self.type]
+            # pylint: disable=protected-access
+            if new_state != self._state:
+                self._state = new_state
+                return True
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def should_poll(self):  # pylint: disable=no-self-use
+        """No polling needed."""
+        return False
+
+    @property
+    def entity_picture(self):
+        """Weather symbol if type is symbol."""
+        if self.type != 'symbol':
+            return None
+        else:
+            return self._entity_picture
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+        }
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return possible sensor specific icon."""
+        if self.type in SENSOR_ICONS:
+            return SENSOR_ICONS[self.type]
+
+
+class BrData(object):
+    """Get the latest data and updates the states."""
+
+    def __init__(self, hass, coordinates, devices):
+        """Initialize the data object."""
+        self._url = 'https://xml.buienradar.nl/'
+        self.devices = devices
+        self.data = {}
+        self.hass = hass
+        self.coordinates = coordinates
+        self.weatherstation = None
+
+    @asyncio.coroutine
+    def update_devices(self):
+        """Update all devices/sensors."""
+        if self.devices:
+            tasks = []
+            # Update all devices
+            for dev in self.devices:
+                if dev.load_data(self.data):
+                    tasks.append(dev.async_update_ha_state())
+
+            if tasks:
+                yield from asyncio.wait(tasks, loop=self.hass.loop)
+
+    @asyncio.coroutine
+    def async_update(self, *_):
+        """Update the data from buienradar."""
+        def get_temp(section):
+            """Get the forecasted temp from xml section."""
+            try:
+                return (float(section[CONF_BR_MINTEMP]) +
+                        float(section[CONF_BR_MAXTEMP])) / 2
+            except (ValueError, TypeError, KeyError):
+                return None
+
+        def select_nearest_ws(xmldata):
+            """Select the nearest weatherstation."""
+            dist = 0
+            loc_data = None
+
+            xmldata_tmp = xmldata
+            if CONF_BR_WEERGEGEVENS not in xmldata_tmp:
```

Comment:
I agree; I modified this while preparing to upload, since I used to have one big IF statement before, but that did not pass the test of the max line length.... :-)

What I can do is test to see of the final path I need exists in the XML file and catch the potential error; something like this:
```
try:
  xmldata_tmp = xmldata[CONF_BR_WEERGEGEVENS][CONF_BR_ACTUEELWEER][CONF_BR_WEERSTATIONS][CONF_BR_WEERSTATION]
except:
  _LOGGER.warning("Missing section in Buienradar xmldata (%s)."
                                "This can happen 00:00-01:00 CE(S)T",
                                CONF_BR_WEERSTATION)
```